spec/system/distribution_system_spec.rb

```diff
@@ -28,6 +28,29 @@
       end
     end
 
+    it "Allows distribution to be created with a time frame" do
+      with_features email_active: true do
+        visit @url_prefix + "/distributions/new"
+
+        select @partner.name, from: "Partner"
+        select @storage_location.name, from: "From storage location"
+
+        fill_in "Comment", with: "Take my wipes... please"
+        check "distribution[issued_at_timeframe_enabled]"
+        select "15", from: "distribution[issued_at_end(5i)]"
+        expect do
+          click_button "Save", match: :first
+        end.to change { ActionMailer::Base.deliveries.count }.by(1)
+
+        expect(page).to have_content "Distributions"
+        expect(Distribution.last.issued_at_timeframe_enabled).to eq(true)
+        expect(Distribution.last.issued_at_end).not_to eq(nil)
+        expect(page.find(".alert-info")).to have_content "reated"
+      end
+      visit(pickup_day_distributions_path(organization_id: @organization.id))
+      page.has_css?("td", text: "PM") or page.has_css?("td", text: "AM")
```

Comment:
Hey there @ccedacero -- I think this is a good opportunity to "fix" the time so that the results will always be the same. In testing, sometimes "brittle" tests that fail sometimes and sometimes not is because there is a dependency on time. By using either https://makandracards.com/makandra/465995-activesupport-includes-timecop-like-helpers or https://github.com/travisjeffery/timecop we can force the time to stay regardless of when you run it.

Knowing that here is what I'd do
- freeze the time by the means mentioned above
- Test the the rendered time is what you froze the time to be.

Let me know if you have any Qs :)

Reply:
Hi Edwin! Thank you for your feedback! 
I'm a bit confused, and I hope you don't mind my question. 

If I just freeze time and compare it to rendered time to make sure they're equal, I don't get how this is testing whether the distribution column has a pickup time. 

Or should I do the following: 

I should freeze the time when a distribution is being created. I should then add the number of days/hours based on the selected distribution date and time?  I can  then verify that this matches the rendered date and time? 

Thank you so much :)